src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/h2/H2EngineSchemaTest.java

```diff
@@ -128,4 +142,37 @@ public static int GetOne() {
     public static int TimesTwo(int value) {
         return value * 2;
     }
+
+    /**
+     * Tests if the timeout setting is properly set in the connection socket to the database.
+     * Note: the DB2 and H2 databases don't support setting the connection socket timeout
+     *
+     * @throws DatabaseFactoryException
+     * @throws ClassNotFoundException
+     * @throws InterruptedException
+     * @throws RecoveryException
+     * @throws RetryLimitExceededException
+     * @throws SQLException
+     */
+    @Test
+    public void testTimeout() throws DatabaseFactoryException, ClassNotFoundException, InterruptedException, RecoveryException, RetryLimitExceededException, SQLException {
+        final int socketTimeout = 60;// in seconds
+        final int loginTimeout = 30;// in seconds
+        final Properties properties = new Properties() {
+            {
+                setProperty(JDBC, config.jdbc);
+                setProperty(USERNAME, config.username);
+                setProperty(PASSWORD, config.password);
+                setProperty(ENGINE, config.engine);
+                setProperty(SCHEMA_POLICY, "create");
+                setProperty(SOCKET_TIMEOUT, Integer.toString(socketTimeout));
+                setProperty(LOGIN_TIMEOUT, Integer.toString(loginTimeout));
+            }
+        };
+
+        final DatabaseEngine de = DatabaseFactory.getConnection(properties);
+        final int connectionTimeoutInMs = de.getConnection().getNetworkTimeout();
+        // Not supported
+        assertEquals("timeout in db connection ok?", 0, connectionTimeoutInMs);
```

Comment:
Error message could be more specific: "Is the timeout of the DB connection disabled as expected?"

Reply:
Yes, that is a special case where the socket timeout can't be set currently for the H2 and DB2 databases. Thanks for the feedback.